src/app/Topology/Actions/NodeActions.tsx

```diff
@@ -139,11 +141,74 @@ export const nodeActions: NodeAction[] = [
   },
   { isSeparator: true },
   {
-    action: (element, { services }, track) => {
+    action: (element, { services }) => {
       const targetNode: TargetNode = element.getData();
-      track(services.api.deleteTarget(targetNode.target).subscribe(() => undefined));
+      services.api.deleteTarget(targetNode.target).subscribe(() => undefined);
     },
     title: 'Delete Target',
     includeList: [NodeType.CUSTOM_TARGET],
   },
+  {
+    action: (element, { services }) => {
+      const group: EnvironmentNode = element.getData();
+      const recordingName = `${group.name.replace(/[\s+-]/g, '_')}_${Date.now()}`;
+      services.api
+        .graphql(
+          `
+            query StartRecordingForGroup($filter: EnvironmentNodeFilterInput) {
+              environmentNodes(filter: $filter) {
+                  descendantTargets {
+                      doStartRecording(recording: {
+                          name: "${recordingName}",
```

Comment:
I *think* this `recordingName` can also be passed to the server as a GraphQL parameter the same way that `$filter` is being done? If so then it's just a little more optimized because the server only needs to parse this query one time and then it knows how to re-execute it with different values for the variables. Doing client-side string interpolation to build the query means it busts the parsed query cache on the server side.